R/zchunk_L251.en_ssp_nonco2.R

```diff
@@ -35,106 +40,214 @@ module_emissions_L251.en_ssp_nonco2_DISABLED <- function(command, ...) {
+       subset(supplysector == "electricity") %>%
+       # Create 2 new columns for future emission factors.
+       # Future emission coefficient year is based on pre-existing "year" column.
+       # Future emission coefficient name is a descriptor, and is constant.
+       mutate(future.emiss.coeff.year = year,
+              future.emiss.coeff.name = "SSP_GAINS",
+              # Previous year column that only includes the electricity supply sector is now a constant.
+              year = 1975)->
+      L251.ssp15_ef_vin
+
+     L251.ssp2_ef %>%
+       # Isolate the "electricity" supply sector.
+       subset(supplysector == "electricity") %>%
+       # Create 2 new columns for future emission factors.
+       # Future emission coefficient year is based on pre-existing "year" column.
+       # Future emission coefficient name is a descriptor, and is constant.
+       mutate(future.emiss.coeff.year = year,
+              future.emiss.coeff.name = "SSP_GAINS",
+              # Previous year column that only includes the electricity supply sector is now a constant.
+              year = 1975)->
+      L251.ssp2_ef_vin
+
+     L251.ssp34_ef %>%
+       # Isolate the "electricity" supply sector.
+       subset(supplysector == "electricity") %>%
+       # Create 2 new columns for future emission factors.
+       # Future emission coefficient year is based on pre-existing "year" column.
+       # Future emission coefficient name is a descriptor, and is constant.
+       mutate(future.emiss.coeff.year = year,
+              future.emiss.coeff.name = "SSP_GAINS",
+              # Previous year column that only includes the electricity supply sector is now a constant.
+              year = 1975)->
+      L251.ssp34_ef_vin
+
+# This section renames SO2 variables so that it has regional SO2 emission species.
+     L251.ctrl.delete <- rename_SO2( L251.ctrl.delete, A_regions, FALSE )
+     L251.ssp15_ef <- rename_SO2( L251.ssp15_ef, A_regions, FALSE )
+     L251.ssp2_ef <- rename_SO2( L251.ssp2_ef, A_regions, FALSE )
+     L251.ssp34_ef <- rename_SO2( L251.ssp34_ef, A_regions, FALSE )
+     L251.ssp15_ef_vin <- rename_SO2( L251.ssp15_ef_vin, A_regions, FALSE )
+     L251.ssp2_ef_vin <- rename_SO2( L251.ssp2_ef_vin, A_regions, FALSE )
+     L251.ssp34_ef_vin <- rename_SO2( L251.ssp34_ef_vin, A_regions, FALSE )
+
+# This section deletes GDP control functions that exist for the supply sector in all
+# regions for non-CO2 emission species for the 1975 control period.
+     L251.ctrl.delete %>%
+       semi_join(L201.nonghg_steepness,
+                 by = c("region", "supplysector", "subsector", "stub.technology", "Non.CO2")) ->
+       L251.ctrl.delete
+
+# ===================================================
+# Produce outputs
+     # No flags are neccessary because old data is in 'long' format.
+
+    L251.ctrl.delete %>%
+      add_title("GDP control of regional non-CO2 emissions.") %>%
+      add_units("NA") %>%
```

Comment:
There are no numbers in this output file.  Everything is in string/character, with the exception of the "year" column (which is a constant).